### PR TITLE
MGMT-6749: Automate ocm-2.3 tag for quay.io/ocpmetal

### PR DESCRIPTION
**Pull request by lranjbar**
# Description

This PR adds published images in quay.io/ocpmetal for the ocm-2.3 branch. It creates a dated image during the midnight run.

**Note: This change is targeted at the ocm-2.3 branch directly without a Bugzilla ticket because it is making changes to the CI automation for release. There are no changes to the assisted code base for QE to verify.**

# What environments does this code impact?

- [ ] Cloud
- [x] Operator Managed Deployments
- [ ] None
- [x] CI

# How was this code tested?

Please, select one or more if needed:

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [x] Waiting for CI to do a full test run
- [x] Manual (Elaborate on how it was tested)
- [ ] No tests needed

I tested the groovy code outside of Jenkins using the groovy playground. Also using the Jenkins replay functionality on our Jenkins server test ran this code with my previous PR #1904. [Jenkins link to PR-1904](http://assisted-jenkins.usersys.redhat.com/job/assisted-service/job/PR-1904/) Also did Jenkins replayed in this PR a couple times as well.

# Assignees

Please, add one or two reviewers that could help review this PR.

/assign @YuviGold
/assign @djzager 

## Checklist

- [X] Title and description added to both, commit and PR.
- [X] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [X] Reviewers have been listed
- [X] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change includes unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

**Review by djzager — APPROVED** [review]

File: Jenkinsfile [hunk 1]
```diff
@@ -1,12 +1,52 @@
-String cron_string = BRANCH_NAME == "master" ? "@hourly" : BRANCH_NAME.startsWith("PR") ? "@midnight" : ""
+// Get current date
+def now = new Date()
+
+// isTriggerByTimer == true if the timer triggers it
+// isEmpty() on lists is currently broken in Jenkins pipelines... so we have to rely on the size of the list
```
Comment (YuviGold) [inline Jenkinsfile:5]: Is there a Jenkins ticket to follow this?

**Review by flaper87 — APPROVED** [review]
Looks good! 

It's unclear to me why this PR is being held, tho.

File: Jenkinsfile (hunk 1 above)
Comment (lranjbar) [reply Jenkinsfile:5]: The root of it is actually: https://issues.jenkins.io/browse/JENKINS-50863 and at some point list.isEmpty() was removed from the  default whitelist. I was following the issue a few years ago myself but gave up on it.

**Pull request merged**